chore: add Prettier and Eslint Checks as GitHub PR Merge Workflows #1640

SumantxD opened the pull request:
<!--   Thank you for your contribution. Before you submit the pull request:
1. Follow our contribution guidelines
2. Test your changes and attach their results to the pull request.
3. Update the relevant documentation.
-->

**Description**
1) added Prettier and ESLint in your project
2) Add a GitHub Actions workflow file

- ...
- ...
- ...

**Related issue(s)**
<!-- If you refer to a particular issue, provide its number, otherwise, remove this section.
For example, `Resolves #123`, `Fixes #43`, or `See also #33`. The `See also #33` option will not automatically close the issue after the PR merge. -->

SumantxD committed:
chore: Add Prettier and Eslint Checks as GitHub PR Merge Workflows #1640

Signed-off-by: SumantxD <st72365@gmail.com>

anshgoyalevil commented:
@SumantxD
Well written code! Kudos! 🛩️ 

Here are some points you might have to consider:
- We need to fix the errors and warning related to eslint and prettier before having this solution in place. Otherwise, all the future PRs would fail the PR merge checks.
- The .eslintrc.js file needs some additional rules according to the project.
- A global eslint npm plugin needs to be added for all AsyncAPI projects.

SumantxD committed:
chore: fixed the errors #1640

Signed-off-by: SumantxD <st72365@gmail.com>

SumantxD commented:
@anshgoyalevil 
* the code has been refactored to fix the eslint and prettier requirements
* additional rules have been added to .eslintrc.js file to meet project's requirements
* here are a list of changes i have made
 1) added 'eslint-plugin-prettier' , This plugin allows ESLint to run Prettier as an ESLint rule. It helps integrate Prettier's code                                                 formatting capabilities into the ESLint workflow, ensuring consistent code style.
 2) added 'eslint-config-prettier' , This configuration disables all ESLint rules that may conflict with Prettier's formatting rules. It allows you to use ESLint and Prettier together without encountering conflicts.
 3) modified '.eslintrc.js' to integrate ESLint and Prettier configuration
 3.1) in env specified that node env is needed
 3.2) modified it to add support for typescript
 3.3) in rules the prettier formatting is enforced
 3.4) the react version has been set to autodetect
 4) refactored the code to make it in favor of ES2015 module syntax

akshatnema commented:
https://github.com/asyncapi/website/issues/1640#issuecomment-1549011700 comment added in the issue.

akshatnema commented:
Closing this PR now since we already resolved the ESLint issue in another PR.
